core/server/common/src/main/java/alluxio/master/journal/Checkpointed.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Base class for Alluxio classes which can be written to and read from metadata checkpoints.
+ */
+public interface Checkpointed {
```

Comment:
net yet, but soon there will be. `RocksInodeStore` is not `Journaled` since we don't write journal entries when we change it, but it will be `Checkpointed` if we support taking `RocksDB-based` checkpoints.